### DISCUSSION TITLE
Memory requirement for GPU KMeans

### DISCUSSION THREAD

**Discussion by VoVAllen** [facebookresearch/faiss#3903] [General]
Hi, I want to use Faiss and GPU to perform KMeans on my own data and get the centroids. What are the gpu and cpu memory requirements for Faiss GPU? I hope to cluster 300GB of data with an L4 (24GB) GPU.

Also could you also suggest some code for me to learn from? There're multiple gpu examples and some seems outdated.

**Comment by mdouze** [comment]
Please point us to outdated examples. 
The clustering training data does not need to reside on GPU, only the centroids need to be on GPU, so for typical number of centroids memory is not a limiting factor.
